File: android/Reguerta/app/src/main/java/com/reguerta/user/domain/shifts/ShiftPlanningInspection.kt
package com.reguerta.user.domain.shifts

import kotlinx.coroutines.flow.Flow

enum class ShiftPlanningMode {
    PREVIEW,
    STAGE,
    ACTIVATE,
}

data class ShiftPlanningSubplanSummary(
    val targetSeasonStartYear: Int,
    val generatedShiftCount: Int,
    val affectedProjectionSeasonStartYears: List<Int>,
)

data class ShiftPlanningCompletedSummary(
    val delivery: ShiftPlanningSubplanSummary,
    val market: ShiftPlanningSubplanSummary,
)

data class ShiftPlanningFailure(
    val scope: String,
    val code: String,
    val messageKey: String,
)

data class ShiftPlanningCandidateReference(
    val candidateId: String,
    val candidateDigest: String,
    val bundleRevision: String,
    val bundleDigest: String,
    val environment: String,
)

data class ShiftPlanningRequestObservation(
    val id: String,
    val bundleId: String,
    val requestedByUserId: String,
    val requestedAtMillis: Long,
    val mode: ShiftPlanningMode,
    val status: ShiftPlanningRequestStatus,
    val completedSummary: ShiftPlanningCompletedSummary?,
    val failure: ShiftPlanningFailure?,
    val candidateReference: ShiftPlanningCandidateReference?,
)

data class ShiftPlanningCandidatePosition(
    val id: String,
    val type: ShiftPlanningRequestType,
    val scheduledDate: String,
    val assignedUserIds: List<String>,
    val helperUserId: String?,
)

data class ShiftPlanningCandidate(
    val id: String,
    val bundleRevision: String,
    val bundleDigest: String,
    val candidateDigest: String,
    val positionDocumentCount: Int,
    val assignmentPositionCount: Int,
    val positions: List<ShiftPlanningCandidatePosition>,
)

interface ShiftPlanningInspectionRepository {
    fun observeLatestRequest(): Flow<ShiftPlanningRequestObservation?>

    suspend fun getStagedCandidate(reference: ShiftPlanningCandidateReference): ShiftPlanningCandidate
}

File: ios/Reguerta/Reguerta/Domain/Shifts/ShiftPlanningInspection.swift
import Foundation

nonisolated enum ShiftPlanningMode: String, Equatable {
    case preview
    case stage
    case activate
}

nonisolated struct ShiftPlanningSubplanSummary: Equatable {
    let targetSeasonStartYear: Int
    let generatedShiftCount: Int
    let affectedProjectionSeasonStartYears: [Int]
}

nonisolated struct ShiftPlanningCompletedSummary: Equatable {
    let bundleRevision: String
    let bundleDigest: String
    let delivery: ShiftPlanningSubplanSummary
    let market: ShiftPlanningSubplanSummary
}

nonisolated struct ShiftPlanningFailure: Equatable {
    let scope: String
    let code: String
    let messageKey: String
}

nonisolated struct ShiftPlanningCandidateReference: Equatable {
    let candidateId: String
    let candidateDigest: String
    let bundleRevision: String
    let bundleDigest: String
    let environment: SessionEnvironment
}

nonisolated struct ShiftPlanningRequestObservation: Identifiable, Equatable {
    let id: String
    let bundleId: String
    let requestedByUserId: String
    let requestedAtMillis: Int64
    let mode: ShiftPlanningMode
    let status: ShiftPlanningRequestStatus
    let completedSummary: ShiftPlanningCompletedSummary?
    let failure: ShiftPlanningFailure?
    let candidateReference: ShiftPlanningCandidateReference?
}

nonisolated struct ShiftPlanningCandidatePosition: Identifiable, Equatable {
    let id: String
    let type: ShiftPlanningRequestType
    let scheduledDate: String
    let assignedUserIds: [String]
    let helperUserId: String?
}

nonisolated struct ShiftPlanningCandidate: Identifiable, Equatable {
    let id: String
    let bundleRevision: String
    let bundleDigest: String
    let candidateDigest: String
    let positionDocumentCount: Int
    let assignmentPositionCount: Int
    let positions: [ShiftPlanningCandidatePosition]
}
